tests/Avalonia.Controls.UnitTests/AppBuilderTests.cs

```diff
@@ -18,6 +19,22 @@ class App : Application
         {
         }
 
+        public class AppWithDependencies : Application
+        {
+            public AppWithDependencies()
```

Comment:
you can have `private AppWithDependencies ()` here rather than an exception... this makes it impossible for anyone to call the default ctor.